sdk/eventhub/event-hubs/test/partitionPump.spec.ts

```diff
@@ -16,4 +30,147 @@ describe("PartitionPump", () => {
     // ...and if they don't override then we give them the default (currently it's EventPosition.earliest())
     getStartingPosition(undefined, undefined).offset!.should.equal(-1);
   });
+
+  describe("telemetry", () => {
+    const eventHubProperties = {
+      endpoint: "theendpoint",
+      eventHubName: "theeventhubname"
+    };
+
+    it("basic span properties are set", async () => {
+      let name: string | undefined;
+      let options: SpanOptions | undefined;
+      const fakeParentSpan = new NoOpSpan();
+      const createdSpan = new SlightlyLessNoOpSpan("myspan");
+
+      const actualReturnedSpan = await createProcessingSpan(
+        [],
+        eventHubProperties,
+        {
+          tracingOptions: {
+            spanOptions: {
+              parent: fakeParentSpan
+            }
+          }
+        },
+        {
+          startSpan: (nameArg, optionsArg) => {
```

Comment:
seems like maybe the other design was for this -- but I would prefer you `setTracer` and object with this `startSpan` method instead if that's what you need.

Reply:
Did a merge + update and all of a sudden there was so much test functionality available in tracing. ;)

I swapped over completely and more or less just eliminated all of my custom code.